src/draft06.rs

```diff
@@ -0,0 +1,756 @@
+//! API for protocol functionality such as creating and parsing ODoH queries and responses.
```

Comment:
I would recommend not using new draft names as filenames because it makes figuring out what changed between two versions of the protocol much harder as Github doesn't allow you to diff between files with different names.

Reply:
My initial thought was, maybe we would have to support multiple draft version at the same time. I agree it will make tracking a bit hard. I can change the filename back to previous one.

Reply:
I don't think supporting multiple versions is very useful, especially since we can do 0-downtime version upgrades now.